pkg/deploy/api/validation/validation.go

```diff
@@ -39,7 +39,18 @@ func ValidateDeploymentConfigSpec(spec deployapi.DeploymentConfigSpec) field.Err
 	if spec.Template == nil {
 		allErrs = append(allErrs, field.Required(specPath.Child("template"), ""))
 	} else {
+		// rather than handleEmptyImageReferences returning an array of changed containers, PR review deemed it more
+		// efficient to build a list of the original container to image mappings, and then just restore all of them (and hence,
+		// even the ones not modified)
+		originalContainerImages := make(map[string]string)
+		for _, container := range spec.Template.Spec.Containers {
+			originalContainerImages[container.Name] = container.Image
+		}
+		handleEmptyImageReferences(spec.Template, spec.Triggers)
```

Comment:
I think he said deep copy was overkill... I think what I'm proposing is similar to what he suggested in https://github.com/openshift/origin/pull/9167#discussion_r65827639